runtime/compiler/env/VMJ9.cpp

```diff
@@ -4115,6 +4115,8 @@ static bool foldFinalFieldsIn(const char *className, int32_t classNameLength, TR
       return true;
    else if (classNameLength >= 38 && !strncmp(className, "java/util/concurrent/ThreadLocalRandom", 38))
       return true;
+   else if (classNameLength >= 25 && !strncmp(className, "java/nio/DirectByteBuffer", 25))
```

Comment:
what analysis have you done to prove that this is ok? enabling HCR should allow the fields to fold using OSR protection so some justification for the whitelisting is warranted.

Reply:
@andrewcraik  Thanks for pointing this out. I did not do any formal analysis on whitelisting DirectByteBuffer class. I am disabling HCR for the workload I am running so I need to test if enabling HCR is fine with the performance or not. Another analysis I did is used a reflection to modify the static final field in the Direct Byte Buffer class. Initial experiments suggests that it is not OK fold final field in this class. Looking into alternate solution to this issue if I have to disable the HCR.

Reply:
Does this mean this pull request should be closed since you will either not need a change or will need a different change?

Reply:
I will need a different change, This PR also contains change to whitelist NOP calls. I was thinking to just remove changes to whilte list DBB from this commit. If you want I will open up separate PR for whitelisting NOP calls and close this one.